pkg/cmd/server/origin/handlers.go

```diff
@@ -168,6 +176,98 @@ func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMa
 	})
 }
 
+// versionSkewFilter adds a filter that may deny PUT an POST requests since they stomp entire resource objects from skewed
+// oc clients, since we know that those clients will strip unknown fields which can lead to unexpected outcomes
+func (c *MasterConfig) versionSkewFilter(binaryInfo version.Info, handler http.Handler) http.Handler {
+	skewedClientPolicy := c.Options.PolicyConfig.SkewedClientPolicy
+	if skewedClientPolicy == configapi.AllowAll {
+		return handler
+	}
+	seg := strings.SplitN(binaryInfo.GitVersion, "-", 2)
```

Comment:
If you need this split it should be done in the build, not here.  This is fragile.